MyCapytain/resources/prototypes/capitains/text.py

```diff
@@ -0,0 +1,137 @@
+from typing import Union
+from MyCapytain.common.reference import URN, BaseReferenceSet, BaseReference
+from MyCapytain.common.constants import RDF_NAMESPACES
+from ..metadata import Collection
+from rdflib import Literal
+from collections import defaultdict
+
+from .collection import CapitainsReadableMetadata
+from ..text import InteractiveTextualNode
+
+
+__all__ = [
+    "PrototypeCapitainsNode",
+    "PrototypeCapitainsText",
+    "PrototypeCapitainsPassage"
+]
+
+
+class PrototypeCapitainsNode(InteractiveTextualNode):
+    """ Initiate a Resource object
+
+    :param urn: A URN identifier
+    :type urn: MyCapytain.common.reference._capitains_cts.URN
+    :param metadata: Collection Information about the Item
+    :type metadata: Collection
+    :param citation: XmlCtsCitation system of the text
+    :type citation: Citation
+    :param children: Current node Children's Identifier
+    :type children: [str]
+    :param parent: Parent of the current node
+    :type parent: str
+    :param siblings: Previous and next node of the current node
+    :type siblings: str
+    :param depth: Depth of the node in the global hierarchy of the text tree
+    :type depth: int
+    :param resource: Resource used to navigate through the textual graph
+
+    :cvar default_exclude: Default exclude for exports
+    """
+
+    @property
+    def urn(self) -> str:
+        """ Kept for backward compatibility
+
+        """
+        return self.id
+
+    def get_capitains_metadata(self, key: str, lang: str = None) -> Union[dict, list]:
+        """ Get easily a metadata from the CAPITAINS namespace
+
+        :param key: CAPITAINS property to retrieve
+        :param lang: Language in which it should be
+        :return: Dictionary in form {lang: [Literal]} if 'lang' is None else a list in form [Literal]
+        """
+        x = defaultdict(list)
+        for obj in self.metadata.get(RDF_NAMESPACES.CAPITAINS.term(key)):
+            x[getattr(obj, 'language', "")].append(obj)
+        if lang is not None:
+            if lang in x:
+                return x[lang]
+            return next(iter(x.values()))
+        return x
+
+    def getValidReff(self, level: int = 1, reference: BaseReference = None) -> BaseReferenceSet:
+        """ Given a resource, CtsText will compute valid reffs
+
+        :param level: Depth required. If not set, should retrieve first encountered level (1 based)
+        :param reference: Subreference (optional)
+        :returns: List of levels
+        """
+        raise NotImplementedError()
+
+    def getLabel(self) -> Collection:
```

Comment:
I think this is CTS dependant, need to check

Reply:
`text.py` is simply a placeholder for testing purposes until the text guidelines are finalized. I won't make any changes to this right now.